lib/core/src/math/utils.rs

```diff
@@ -56,3 +112,8 @@ pub fn invert_y(mut m: Matrix4<f32>) -> Matrix4<f32> {
     m.row_part_mut(1, 4).iter_mut().for_each(|a| *a = -*a);
     m
 }
+
+// Note [2D array to 1D array]
+// ===========================
+// We need to interpret [[f32; 4]; 4] as [f32; 16] to view as Float32Array.
+// The memory layout is the same, so this operation is safe.
```

Comment:
Imo this comment on unsafe function is still misleading. It tells it is safe, while the function is marked unsafe.

Reply:
In this function there are two different unsafe operations: raw pointer casting and the use of `Float32Array::view`. What if I group here? Will it be more evident?

```rust
impl IntoFloat32ArrayView for Matrix4<f32> {
    unsafe fn into_float32_array_view(&self) -> Float32ArrayView<'_> {
        // Note [2D array to 1D array]
        let matrix = {
            let matrix = self.as_ref();
            let matrix = matrix as *const [[f32; 4]; 4];
            &*(matrix as *const [f32; 16]);
        };
        let array = Float32Array::view(matrix);
        let phantom = PhantomData;
        Float32ArrayView { array, phantom }
    }
}
```

Reply:
I was refering to the sentence `We need to interpret [[f32; 4]; 4] as [f32; 16] to view as Float32Array. The memory layout is the same, so this operation is safe.` - which is not really true, as the whole operation is marked unsafe. If the whole operation is marked unsafe, we should not have a comment telling "hey, this is safe".

Reply:
As described in the comment bellow, this section of code was wrapped into a safe function called `matrix4_to_array`.